### DISCUSSION TITLE
Is it possible to call a Javascript function that returns a value from native code?

### DISCUSSION THREAD

**Discussion by sundeepgupta** [ionic-team/capacitor#3937] [Q&A] [answered]
I'm working on writing tests across the bridge and I see that native code can trigger events that the JS can listen for. However, I'd like for the native code to get a return value back from the JS code. If this is not possible, is there a somewhat elegant way to achieve this?

**Comment by crixx** [comment, marked as answer]
From native you can inject JS into the web-view, which then is executed there.

```    
// just an example
func showSplashScreen(){
        self.bridge.getWebView()?.evaluateJavaScript("Capacitor.Plugins.SplashScreen.show({showDuration: 4000, autoHide: true});", completionHandler: nil)
}
```

I think it should then be possible to gather the data there and return them, such that the completion handler fires. Take into consideration that this is an async operation...

And other way could be to provide something like a DataSink in your plugin API, such that you can gather the data and then send it to the DataSink...

```    

func triggerDataLoad(){       self.bridge.getWebView()?.evaluateJavaScript("Capacitor.Plugins.YourDataSinkPlugin.swallow(window.YOUR_GLOBAL.gatherData());", completionHandler: nil)
}
```

Does that help anyhow? 🙃 
Crixx

**Reply by sundeepgupta** [reply]
Yes it does - thanks
